packages/fxa-settings/src/lib/channels/firefox.ts

```diff
@@ -298,6 +311,42 @@ export class Firefox extends EventTarget {
   fxaCanLinkAccount(options: FxACanLinkAccount) {
     this.send(FirefoxCommand.Login, options);
   }
+
+  async requestSignedInUser(context: string) {
+    return Promise.race([
+      new Promise<void>((resolve) => {
+        const handleFxAStatusEvent = (event: any) => {
+          const status = event.detail as FxAStatusResponse;
+          const signedInUser = status.signedInUser as StoredAccountData;
+          if (signedInUser) {
+            storeAccountData(signedInUser);
+            cache.writeQuery({
+              query: GET_LOCAL_SIGNED_IN_STATUS,
+              data: { isSignedIn: true },
+            });
+          }
+          this.removeEventListener(
+            FirefoxCommand.FxAStatus,
+            handleFxAStatusEvent
+          );
+          resolve();
+        };
+
+        this.addEventListener(FirefoxCommand.FxAStatus, handleFxAStatusEvent);
+        // requestAnimationFrame ensures the event listener is added first
+        // otherwise, there is a race condition
+        requestAnimationFrame(() => {
+          this.send(FirefoxCommand.FxAStatus, {
+            context,
+            isPairing: false,
```

Comment:
@tarikeshaq I didn't include `service` since it [looks like this should always be undefined](https://github.com/mozilla/fxa/pull/15723/files#diff-1d01b521a75c5ddc2f4221af840c294be4ddb23d6d7206d31101d5595bbdea41R82) since this is only called for the OAuth Sync flow (happy to explicitly add this though with a comment), and I included `isPairing: false` since it was included in Vijay's PR. LMK if any of this should be updated.

Reply:
Again great callout! [In Desktop we try to narrow down which cases we allow FXA_STATUS in private mode, and we use the `service=sync` as an indicator that we should keep it](https://searchfox.org/mozilla-central/rev/f465027ef4d334dbc9ad270718c8a5e8045e7a2b/services/fxaccounts/FxAccountsWebChannel.sys.mjs#515-538).. However, there is a context check there too.

I say we *don't* send it, and when enabling support for oauth in desktop I can have Desktop check if the context is `oauth_webchannel_v1` if oauth is enabled